src/value/value.h

```diff
@@ -133,37 +133,19 @@ typedef struct RSValue {
 // Constructors
 ///////////////////////////////////////////////////////////////
 
-#ifndef __cplusplus
 /**
- * Creates a stack-allocated undefined RSValue.
- * The returned value is not allocated on the heap and should not be freed.
- * @return A stack-allocated RSValue of type RSValueType_Undef
- */
-RSValue RSValue_Undefined();
-
-/**
- * Creates a stack-allocated RSValue containing a number.
- * The returned value is not allocated on the heap and should not be freed.
- * @param n The numeric value to wrap
- * @return A stack-allocated RSValue of type RSValueType_Number
- */
-RSValue RSValue_Number(double n);
-
-/**
- * Creates a stack-allocated RSValue containing a malloc'd string.
- * The returned value itself is not heap-allocated, but takes ownership of the string.
- * @param str The malloc'd string to wrap (ownership is transferred)
- * @param len The length of the string
- * @return A stack-allocated RSValue of type RSValue_String with RSString_Malloc subtype
+ * Creates a heap-allocated Undefined RSValue.
+ * @return A pointer to a heap-allocated RSValue
  */
-RSValue RSValue_String(char *str, uint32_t len);
-#endif
+RSValue *RSValue_NewUndefined();
```

Comment:
Not sure if the fix is part of this PR, but there are some comments where we are using `RSValue_String` where it should be `RSValue_TypeString`